py/nlrthumb.c

```diff
@@ -76,9 +76,9 @@ __attribute__((naked)) unsigned int nlr_push(nlr_buf_t *nlr) {
 #endif
     );
 
-    #if defined(__GNUC__)
+    #if __GNUC__ < 4 || (__GNUC__ == 4 && __GNUC_MINOR__ < 8)
```

Comment:
I just noticed that this will probably not work well with non-GNUC compilers, because the `__GNUC__ < 4` check will probably succeed if `__GNUC__` is not defined.  So I guess it needs to retain the `defined(__GNUC__) bit at the start of the if.

Reply:
Ah, yes, that check should be there. Fixed it.